include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+                              std::string const & argument_value)
+    {
+
+        ctd_stream << "\t\t\t<ITEM "
+                   << "name=\"" << argument_name << "\" "
+                   << "type=\"" << argument_type << "\" "
+                   << "description=\"" << argument_description << "\" "
+                   << "restrictions=\"" << argument_restrictions << "\" "
+                   << "required=\"" << argument_required << "\" "
+                   << "advanced=\"" << argument_advanced << "\" "
+                   << "value=\"" << argument_value << "\"";
+        if (argument_type == std::string{"input-file"} || 
+            argument_type == std::string{"output-file"} || 
+            argument_type == std::string{"input-prefix"} || 
+            argument_type == std::string{"output-prefix"})
+        {
+            ctd_stream << " supported_formats=\"" << argument_formats << "\"";
+        }
+        ctd_stream << "/>\n";
+    }
+
+    /*!\brief Append the CTD 'NODE' subtree to the current CTD stream buffer.
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_node_node(argument_parser_meta_data const & meta) 
+    {
+        ctd_stream << "\t\t<NODE " 
+                   << "name=\"" << meta.app_name << "\" "
+                   << "description=\"" << escape_special_xml_chars(meta.short_description) << "\""
+                   << ">\n"; 
+        for (auto f : item_option_callbacks)
+        {
+            f();
+        }
+        for (auto f : item_argument_callbacks)
+        {
+            f();
+        }
+        ctd_stream << "\t\t</NODE>\n";
+    }
+
+    /*!\brief Append the CTD 'PARAMETERS' subtree to the current CTD stream buffer.
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_parameters_node(argument_parser_meta_data const & meta) 
+    {
+        ctd_stream << "\t<PARAMETERS version=\"1.7.0\">\n";
+        append_ctd_node_node(meta);
+        ctd_stream << "\t</PARAMETERS>\n";
+    }
+
+    /*!\brief Allocate and append a CTD 'tool' subtree to the current DOM tree. 
+     *
+     * \param[in] meta A structure storing application meta-data.
+     */
+    void append_ctd_tool_node(argument_parser_meta_data const & meta) 
+    {
+        ctd_stream << "<tool " 
+                   << "name=\"" << meta.app_name << "\" "
+                   << "version=\"" << (meta.version.empty() ? "0.0.0.0" : meta.version) << "\" "
+                   << "docurl=\"" << (meta.url.empty() ? "" : meta.url) << "\" "
+                   << "ctdVersion=\"1.7.0\""
+                   << ">\n";
+        append_ctd_description_node(meta);
+        append_ctd_manual_node(meta);
+        append_ctd_cli_node(meta);
+        append_ctd_parameters_node(meta);
+        ctd_stream << "</tool>\n";
+    }
+
+    //! Stream storing the CTD file lines whenever command line parsing is triggered.
```

Comment:
the `\brief` is missing for all members

Reply:
Ok, as long as the variable member should be self-explanatory I will leave the documentation as it is, just adding the `\brief` tag, avoiding useless details.